spec/api-browser-window-spec.js

```diff
@@ -753,6 +753,72 @@ describe('BrowserWindow module', () => {
     })
   })
 
+  describe('BrowserWindow.setWindowControlsVisibility()', () => {
+    /**
+     * FIXME: Skip the tests instead of using the `return` here.
+     * - `.skip()` called in the 'before' hook doesn't affect
+     *     nested `describe`s.
+     * - Bad support of async test cases
+     */
+    if (process.platform !== 'darwin') {
+      return
+    }
+
+    it('does not throw', () => {
+      assert.doesNotThrow(() => {
+        w.setWindowControlsVisibility(true)
+      })
+    })
+
+    // Checks that window controls visibility behaviour remains the same in simple fullscreen mode
+    // if user did not specified it explicitly.
+    describe('when fullscreen titlebar on', () => {
+      beforeEach(() => {
+        w.destroy()
+        w = new BrowserWindow({ fullscreenWindowTitle: true })
+      })
+
+      it('hides controls when enters simple fullscreen mode', () => {
+        // BUG in SimpleFullScreen implementation? It does not trigger enter/leave-full-screen events
```

Comment:
Can you please create an issue if there is a problem with these events?
https://github.com/electron/electron/issues/new/choose